### ISSUE TITLE
Command org-wc-count-subtrees report error

### ISSUE THREAD

**Issue by stardiviner** [tesujimath/org-wc#23]
Here is the stacktrace:
```
Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  count-words(nil nil)
  (cond ((not (called-interactively-p 'any)) (count-words start end)) (arg (count-words--buffer-message)) (t (count-words--message "Region" start end)))
  count-words-region(nil nil)
  (+ wc (count-words-region (match-beginning 3) (match-end 3)))
  (setq wc (+ wc (count-words-region (match-beginning 3) (match-end 3))))
  (cl-incf wc (count-words-region (match-beginning 3) (match-end 3)))
  (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end))
  (cond ((eql temp 'ignore) (org-wc--goto-char (match-end 0) end)) ((eql temp 'oneword) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) ((eql temp 'description) (if (match-beginning 5) (goto-char (match-beginning 5)) (org-wc--goto-char (match-end 0) end))) ((eql temp 'path) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end)) ((eql temp 'description-or-path) (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end))) (t (user-error "Error in org-wc link configuration")))
  (let* ((temp (cond ((member type org-wc-ignored-link-types) 'ignore) ((member type org-wc-one-word-link-types) 'oneword) ((member type org-wc-only-description-link-types) 'description) ((member type org-wc-only-path-link-types) 'path) ((member type org-wc-description-or-path-link-types) 'description-or-path) (t org-wc-default-link-count)))) (cond ((eql temp 'ignore) (org-wc--goto-char (match-end 0) end)) ((eql temp 'oneword) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) ((eql temp 'description) (if (match-beginning 5) (goto-char (match-beginning 5)) (org-wc--goto-char (match-end 0) end))) ((eql temp 'path) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end)) ((eql temp 'description-or-path) (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end))) (t (user-error "Error in org-wc link configuration"))))
  (cl-case (cond ((member type org-wc-ignored-link-types) 'ignore) ((member type org-wc-one-word-link-types) 'oneword) ((member type org-wc-only-description-link-types) 'description) ((member type org-wc-only-path-link-types) 'path) ((member type org-wc-description-or-path-link-types) 'description-or-path) (t org-wc-default-link-count)) (ignore (org-wc--goto-char (match-end 0) end)) (oneword (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (description (if (match-beginning 5) (goto-char (match-beginning 5)) (org-wc--goto-char (match-end 0) end))) (path (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end)) (description-or-path (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end))) (t (user-error "Error in org-wc link configuration")))
  (let* ((type (match-string 2))) (cl-case (cond ((member type org-wc-ignored-link-types) 'ignore) ((member type org-wc-one-word-link-types) 'oneword) ((member type org-wc-only-description-link-types) 'description) ((member type org-wc-only-path-link-types) 'path) ((member type org-wc-description-or-path-link-types) 'description-or-path) (t org-wc-default-link-count)) (ignore (org-wc--goto-char (match-end 0) end)) (oneword (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (description (if (match-beginning 5) (goto-char (match-beginning 5)) (org-wc--goto-char (match-end 0) end))) (path (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end)) (description-or-path (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end))) (t (user-error "Error in org-wc link configuration"))))
  (cond ((org-wc-in-heading-line) (if (or (and org-wc-ignore-commented-trees (org-in-commented-heading-p)) (cl-intersection org-wc-ignored-tags (org-get-tags-at) :test #'string=)) (org-end-of-subtree t t) (org-wc--goto-char (point-at-eol) end))) ((when (save-excursion (beginning-of-line 1) (looking-at org-block-regexp)) (if (member (match-string 1) org-wc-blocks-to-count) (progn (org-wc--goto-char (match-beginning 4) end) (cl-decf wc)) (org-wc--goto-char (match-end 0) end)))) ((org-at-comment-p) (org-wc--goto-char (point-at-eol) end)) ((org-at-drawer-p) (org-end-of-meta-data t)) ((save-excursion (when (< (1+ (point-min)) (point)) (backward-char 2)) (looking-at org-bracket-link-analytic-regexp)) (let* ((type (match-string 2))) (cl-case (cond ((member type org-wc-ignored-link-types) 'ignore) ((member type org-wc-one-word-link-types) 'oneword) ((member type org-wc-only-description-link-types) 'description) ((member type org-wc-only-path-link-types) 'path) ((member type org-wc-description-or-path-link-types) 'description-or-path) (t org-wc-default-link-count)) (ignore (org-wc--goto-char (match-end 0) end)) (oneword (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (description (if (match-beginning 5) (goto-char (match-beginning 5)) (org-wc--goto-char (match-end 0) end))) (path (cl-incf wc (count-words-region (match-beginning 3) (match-end 3))) (org-wc--goto-char (match-end 0) end)) (description-or-path (if (match-beginning 5) (goto-char (match-beginning 5)) (cl-incf wc (count-words-region ... ...)) (org-wc--goto-char (match-end 0) end))) (t (user-error "Error in org-wc link configuration"))))) ((save-excursion (when (< (point-min) (point)) (backward-char)) (looking-at latex-macro-regexp)) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (t (and (re-search-forward "\\w+\\W*" end 'skip) (cl-incf wc))))
  (while (< (point) end) (cond ((org-wc-in-heading-line) (if (or (and org-wc-ignore-commented-trees (org-in-commented-heading-p)) (cl-intersection org-wc-ignored-tags (org-get-tags-at) :test #'string=)) (org-end-of-subtree t t) (org-wc--goto-char (point-at-eol) end))) ((when (save-excursion (beginning-of-line 1) (looking-at org-block-regexp)) (if (member (match-string 1) org-wc-blocks-to-count) (progn (org-wc--goto-char (match-beginning 4) end) (cl-decf wc)) (org-wc--goto-char (match-end 0) end)))) ((org-at-comment-p) (org-wc--goto-char (point-at-eol) end)) ((org-at-drawer-p) (org-end-of-meta-data t)) ((save-excursion (when (< (1+ (point-min)) (point)) (backward-char 2)) (looking-at org-bracket-link-analytic-regexp)) (let* ((type (match-string 2))) (cl-case (cond ((member type org-wc-ignored-link-types) 'ignore) ((member type org-wc-one-word-link-types) 'oneword) ((member type org-wc-only-description-link-types) 'description) ((member type org-wc-only-path-link-types) 'path) ((member type org-wc-description-or-path-link-types) 'description-or-path) (t org-wc-default-link-count)) (ignore (org-wc--goto-char (match-end 0) end)) (oneword (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (description (if (match-beginning 5) (goto-char ...) (org-wc--goto-char ... end))) (path (cl-incf wc (count-words-region ... ...)) (org-wc--goto-char (match-end 0) end)) (description-or-path (if (match-beginning 5) (goto-char ...) (cl-incf wc ...) (org-wc--goto-char ... end))) (t (user-error "Error in org-wc link configuration"))))) ((save-excursion (when (< (point-min) (point)) (backward-char)) (looking-at latex-macro-regexp)) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (t (and (re-search-forward "\\w+\\W*" end 'skip) (cl-incf wc)))))
  (save-excursion (goto-char beg) (when (org-at-drawer-p) (org-end-of-meta-data t)) (while (< (point) end) (cond ((org-wc-in-heading-line) (if (or (and org-wc-ignore-commented-trees (org-in-commented-heading-p)) (cl-intersection org-wc-ignored-tags (org-get-tags-at) :test #'string=)) (org-end-of-subtree t t) (org-wc--goto-char (point-at-eol) end))) ((when (save-excursion (beginning-of-line 1) (looking-at org-block-regexp)) (if (member (match-string 1) org-wc-blocks-to-count) (progn (org-wc--goto-char ... end) (cl-decf wc)) (org-wc--goto-char (match-end 0) end)))) ((org-at-comment-p) (org-wc--goto-char (point-at-eol) end)) ((org-at-drawer-p) (org-end-of-meta-data t)) ((save-excursion (when (< (1+ ...) (point)) (backward-char 2)) (looking-at org-bracket-link-analytic-regexp)) (let* ((type (match-string 2))) (cl-case (cond (... ...) (... ...) (... ...) (... ...) (... ...) (t org-wc-default-link-count)) (ignore (org-wc--goto-char ... end)) (oneword (org-wc--goto-char ... end) (cl-incf wc)) (description (if ... ... ...)) (path (cl-incf wc ...) (org-wc--goto-char ... end)) (description-or-path (if ... ... ... ...)) (t (user-error "Error in org-wc link configuration"))))) ((save-excursion (when (< (point-min) (point)) (backward-char)) (looking-at latex-macro-regexp)) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (t (and (re-search-forward "\\w+\\W*" end 'skip) (cl-incf wc))))))
  (let ((wc 0) (latex-macro-regexp "\\\\[A-Za-z]+\\(\\[[^]]*\\]\\|\\){\\([^}]*\\)}")) (save-excursion (goto-char beg) (when (org-at-drawer-p) (org-end-of-meta-data t)) (while (< (point) end) (cond ((org-wc-in-heading-line) (if (or (and org-wc-ignore-commented-trees ...) (cl-intersection org-wc-ignored-tags ... :test ...)) (org-end-of-subtree t t) (org-wc--goto-char (point-at-eol) end))) ((when (save-excursion (beginning-of-line 1) (looking-at org-block-regexp)) (if (member ... org-wc-blocks-to-count) (progn ... ...) (org-wc--goto-char ... end)))) ((org-at-comment-p) (org-wc--goto-char (point-at-eol) end)) ((org-at-drawer-p) (org-end-of-meta-data t)) ((save-excursion (when (< ... ...) (backward-char 2)) (looking-at org-bracket-link-analytic-regexp)) (let* ((type ...)) (cl-case (cond ... ... ... ... ... ...) (ignore ...) (oneword ... ...) (description ...) (path ... ...) (description-or-path ...) (t ...)))) ((save-excursion (when (< ... ...) (backward-char)) (looking-at latex-macro-regexp)) (org-wc--goto-char (match-end 0) end) (cl-incf wc)) (t (and (re-search-forward "\\w+\\W*" end 'skip) (cl-incf wc)))))) wc)
  org-word-count-aux(19236 19302)
  (let ((wc (org-word-count-aux (point-min) (point-max)))) (put-text-property (point) (point-at-eol) :org-wc wc) (goto-char (point-min)))
  (save-restriction (org-narrow-to-subtree) (let ((wc (org-word-count-aux (point-min) (point-max)))) (put-text-property (point) (point-at-eol) :org-wc wc) (goto-char (point-min))))
  (while (outline-previous-heading) (save-restriction (org-narrow-to-subtree) (let ((wc (org-word-count-aux (point-min) (point-max)))) (put-text-property (point) (point-at-eol) :org-wc wc) (goto-char (point-min)))))
  (save-excursion (goto-char (point-max)) (while (outline-previous-heading) (save-restriction (org-narrow-to-subtree) (let ((wc (org-word-count-aux (point-min) (point-max)))) (put-text-property (point) (point-at-eol) :org-wc wc) (goto-char (point-min))))))
  org-wc-count-subtrees()
  funcall-interactively(org-wc-count-subtrees)
  call-interactively(org-wc-count-subtrees record nil)
  command-execute(org-wc-count-subtrees record)
  counsel-M-x-action("org-wc-count-subtrees")
  ivy-call()
  #f(compiled-function (arg1 arg2 &rest rest) "Read a string in the minibuffer, with completion.\n\nPROMPT is a string, normally ending in a colon and a space.\n`ivy-count-format' is prepended to PROMPT during completion.\n\nCOLLECTION is either a list of strings, a function, an alist, or\na hash table, supplied for `minibuffer-completion-table'.\n\nPREDICATE is applied to filter out the COLLECTION immediately.\nThis argument is for compatibility with `completing-read'.\n\nWhen REQUIRE-MATCH is non-nil, only members of COLLECTION can be\nselected.\n\nIf INITIAL-INPUT is non-nil, then insert that input in the\nminibuffer initially.\n\nHISTORY is a name of a variable to hold the completion session\nhistory.\n\nKEYMAP is composed with `ivy-minibuffer-map'.\n\nPRESELECT, when non-nil, determines which one of the candidates\nmatching INITIAL-INPUT to select initially.  An integer stands\nfor the position of the desired candidate in the collection,\ncounting from zero.  Otherwise, use the first occurrence of\nPRESELECT in the collection.  Comparison is first done with\n`equal'.  If that fails, and when applicable, match PRESELECT as\na regular expression.\n\nDEF is for compatibility with `completing-read'.\n\nUPDATE-FN is called each time the candidate list is re-displayed.\n\nWhen SORT is non-nil, `ivy-sort-functions-alist' determines how\nto sort candidates before displaying them.\n\nACTION is a function to call after selecting a candidate.\nIt takes one argument, the selected candidate. If COLLECTION is\nan alist, the argument is a cons cell, otherwise it's a string.\n\nMULTI-ACTION, when non-nil, is called instead of ACTION when\nthere are marked candidates. It takes the list of candidates as\nits only argument. When it's nil, ACTION is called on each marked\ncandidate.\n\nUNWIND is a function of no arguments to call before exiting.\n\nRE-BUILDER is a function transforming input text into a regex\npattern.\n\nMATCHER is a function which can override how candidates are\nfiltered based on user input.  It takes a regex pattern and a\nlist of candidates, and returns the list of matching candidates.\n\nDYNAMIC-COLLECTION is a boolean specifying whether the list of\ncandidates is updated after each input by calling COLLECTION.\n\nEXTRA-PROPS can be used to store collection-specific\nsession-specific data.\n\nCALLER is a symbol to uniquely identify the caller to `ivy-read'.\nIt is used, along with COLLECTION, to determine which\ncustomizations apply to the current completion session." #<bytecode 0x1987a553bed4ebb0>)("M-x " [eww-buffer-select helm-ff-avfs-directory diary-included-files epg-context-signers haskell-indentation-bird-p dap-ui-breakpoint-log-message nxml-fontify-matcher eww-handle-link gdb-script-syntax-propertize-function reftex-default-bibliography webmail-encode-8bit send-off treemacs-project->path-status gnus-sync haskell-cabal-empty-regexp ns-aliases helm-type-timers-actions treemacs-next-neighbour none-but-delete xsdre-parse-char-class lsp-ui-flycheck chart-axis--eieio-childp lsp-solargraph-references delphi-column-of haskell-mode--inhibit-bracket-inside-comment-or-default semantic--tag-unlink-list-from-buffer lsp-pyls-plugins-flake8-max-line-length ellipses c-standard-font-lock-fontify-region-function bibtex-SuppCollection calc-arccosh tramp-cleanup xsdre-to-symbolic shr-parse-image-data soap-element-namespace-tag--cmacro org-man-fixed-width treemacs-workspace-p--cmacro helm-kill-regexp-as-sexp helm-browse-project-build-buffers-source \' ---5 helm-grep-mode-jump-other-window-forward helm-buffer-run-ediff-merge ---8 treemacs--follow-after-buffer-list-update batch-update-authors treemacs--follow-tag-at-point helm-buffers--get-tab-from-name \0 tramp-gw-open-connection ...] :predicate #f(compiled-function (sym) #<bytecode 0xcf872e21786708c>) :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x)
  apply(#f(compiled-function (arg1 arg2 &rest rest) "Read a string in the minibuffer, with completion.\n\nPROMPT is a string, normally ending in a colon and a space.\n`ivy-count-format' is prepended to PROMPT during completion.\n\nCOLLECTION is either a list of strings, a function, an alist, or\na hash table, supplied for `minibuffer-completion-table'.\n\nPREDICATE is applied to filter out the COLLECTION immediately.\nThis argument is for compatibility with `completing-read'.\n\nWhen REQUIRE-MATCH is non-nil, only members of COLLECTION can be\nselected.\n\nIf INITIAL-INPUT is non-nil, then insert that input in the\nminibuffer initially.\n\nHISTORY is a name of a variable to hold the completion session\nhistory.\n\nKEYMAP is composed with `ivy-minibuffer-map'.\n\nPRESELECT, when non-nil, determines which one of the candidates\nmatching INITIAL-INPUT to select initially.  An integer stands\nfor the position of the desired candidate in the collection,\ncounting from zero.  Otherwise, use the first occurrence of\nPRESELECT in the collection.  Comparison is first done with\n`equal'.  If that fails, and when applicable, match PRESELECT as\na regular expression.\n\nDEF is for compatibility with `completing-read'.\n\nUPDATE-FN is called each time the candidate list is re-displayed.\n\nWhen SORT is non-nil, `ivy-sort-functions-alist' determines how\nto sort candidates before displaying them.\n\nACTION is a function to call after selecting a candidate.\nIt takes one argument, the selected candidate. If COLLECTION is\nan alist, the argument is a cons cell, otherwise it's a string.\n\nMULTI-ACTION, when non-nil, is called instead of ACTION when\nthere are marked candidates. It takes the list of candidates as\nits only argument. When it's nil, ACTION is called on each marked\ncandidate.\n\nUNWIND is a function of no arguments to call before exiting.\n\nRE-BUILDER is a function transforming input text into a regex\npattern.\n\nMATCHER is a function which can override how candidates are\nfiltered based on user input.  It takes a regex pattern and a\nlist of candidates, and returns the list of matching candidates.\n\nDYNAMIC-COLLECTION is a boolean specifying whether the list of\ncandidates is updated after each input by calling COLLECTION.\n\nEXTRA-PROPS can be used to store collection-specific\nsession-specific data.\n\nCALLER is a symbol to uniquely identify the caller to `ivy-read'.\nIt is used, along with COLLECTION, to determine which\ncustomizations apply to the current completion session." #<bytecode 0x1987a553bed4ebb0>) ("M-x " [eww-buffer-select helm-ff-avfs-directory diary-included-files epg-context-signers haskell-indentation-bird-p dap-ui-breakpoint-log-message nxml-fontify-matcher eww-handle-link gdb-script-syntax-propertize-function reftex-default-bibliography webmail-encode-8bit send-off treemacs-project->path-status gnus-sync haskell-cabal-empty-regexp ns-aliases helm-type-timers-actions treemacs-next-neighbour none-but-delete xsdre-parse-char-class lsp-ui-flycheck chart-axis--eieio-childp lsp-solargraph-references delphi-column-of haskell-mode--inhibit-bracket-inside-comment-or-default semantic--tag-unlink-list-from-buffer lsp-pyls-plugins-flake8-max-line-length ellipses c-standard-font-lock-fontify-region-function bibtex-SuppCollection calc-arccosh tramp-cleanup xsdre-to-symbolic shr-parse-image-data soap-element-namespace-tag--cmacro org-man-fixed-width treemacs-workspace-p--cmacro helm-kill-regexp-as-sexp helm-browse-project-build-buffers-source \' ---5 helm-grep-mode-jump-other-window-forward helm-buffer-run-ediff-merge ---8 treemacs--follow-after-buffer-list-update batch-update-authors treemacs--follow-tag-at-point helm-buffers--get-tab-from-name \0 tramp-gw-open-connection ...] :predicate #f(compiled-function (sym) #<bytecode 0xcf872e21786708c>) :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x))
  ivy-posframe--read(#f(compiled-function (arg1 arg2 &rest rest) "Read a string in the minibuffer, with completion.\n\nPROMPT is a string, normally ending in a colon and a space.\n`ivy-count-format' is prepended to PROMPT during completion.\n\nCOLLECTION is either a list of strings, a function, an alist, or\na hash table, supplied for `minibuffer-completion-table'.\n\nPREDICATE is applied to filter out the COLLECTION immediately.\nThis argument is for compatibility with `completing-read'.\n\nWhen REQUIRE-MATCH is non-nil, only members of COLLECTION can be\nselected.\n\nIf INITIAL-INPUT is non-nil, then insert that input in the\nminibuffer initially.\n\nHISTORY is a name of a variable to hold the completion session\nhistory.\n\nKEYMAP is composed with `ivy-minibuffer-map'.\n\nPRESELECT, when non-nil, determines which one of the candidates\nmatching INITIAL-INPUT to select initially.  An integer stands\nfor the position of the desired candidate in the collection,\ncounting from zero.  Otherwise, use the first occurrence of\nPRESELECT in the collection.  Comparison is first done with\n`equal'.  If that fails, and when applicable, match PRESELECT as\na regular expression.\n\nDEF is for compatibility with `completing-read'.\n\nUPDATE-FN is called each time the candidate list is re-displayed.\n\nWhen SORT is non-nil, `ivy-sort-functions-alist' determines how\nto sort candidates before displaying them.\n\nACTION is a function to call after selecting a candidate.\nIt takes one argument, the selected candidate. If COLLECTION is\nan alist, the argument is a cons cell, otherwise it's a string.\n\nMULTI-ACTION, when non-nil, is called instead of ACTION when\nthere are marked candidates. It takes the list of candidates as\nits only argument. When it's nil, ACTION is called on each marked\ncandidate.\n\nUNWIND is a function of no arguments to call before exiting.\n\nRE-BUILDER is a function transforming input text into a regex\npattern.\n\nMATCHER is a function which can override how candidates are\nfiltered based on user input.  It takes a regex pattern and a\nlist of candidates, and returns the list of matching candidates.\n\nDYNAMIC-COLLECTION is a boolean specifying whether the list of\ncandidates is updated after each input by calling COLLECTION.\n\nEXTRA-PROPS can be used to store collection-specific\nsession-specific data.\n\nCALLER is a symbol to uniquely identify the caller to `ivy-read'.\nIt is used, along with COLLECTION, to determine which\ncustomizations apply to the current completion session." #<bytecode 0x1987a553bed4ebb0>) "M-x " [eww-buffer-select helm-ff-avfs-directory diary-included-files epg-context-signers haskell-indentation-bird-p dap-ui-breakpoint-log-message nxml-fontify-matcher eww-handle-link gdb-script-syntax-propertize-function reftex-default-bibliography webmail-encode-8bit send-off treemacs-project->path-status gnus-sync haskell-cabal-empty-regexp ns-aliases helm-type-timers-actions treemacs-next-neighbour none-but-delete xsdre-parse-char-class lsp-ui-flycheck chart-axis--eieio-childp lsp-solargraph-references delphi-column-of haskell-mode--inhibit-bracket-inside-comment-or-default semantic--tag-unlink-list-from-buffer lsp-pyls-plugins-flake8-max-line-length ellipses c-standard-font-lock-fontify-region-function bibtex-SuppCollection calc-arccosh tramp-cleanup xsdre-to-symbolic shr-parse-image-data soap-element-namespace-tag--cmacro org-man-fixed-width treemacs-workspace-p--cmacro helm-kill-regexp-as-sexp helm-browse-project-build-buffers-source \' ---5 helm-grep-mode-jump-other-window-forward helm-buffer-run-ediff-merge ---8 treemacs--follow-after-buffer-list-update batch-update-authors treemacs--follow-tag-at-point helm-buffers--get-tab-from-name \0 tramp-gw-open-connection ...] :predicate #f(compiled-function (sym) #<bytecode 0xcf872e21786708c>) :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x)
  apply(ivy-posframe--read #f(compiled-function (arg1 arg2 &rest rest) "Read a string in the minibuffer, with completion.\n\nPROMPT is a string, normally ending in a colon and a space.\n`ivy-count-format' is prepended to PROMPT during completion.\n\nCOLLECTION is either a list of strings, a function, an alist, or\na hash table, supplied for `minibuffer-completion-table'.\n\nPREDICATE is applied to filter out the COLLECTION immediately.\nThis argument is for compatibility with `completing-read'.\n\nWhen REQUIRE-MATCH is non-nil, only members of COLLECTION can be\nselected.\n\nIf INITIAL-INPUT is non-nil, then insert that input in the\nminibuffer initially.\n\nHISTORY is a name of a variable to hold the completion session\nhistory.\n\nKEYMAP is composed with `ivy-minibuffer-map'.\n\nPRESELECT, when non-nil, determines which one of the candidates\nmatching INITIAL-INPUT to select initially.  An integer stands\nfor the position of the desired candidate in the collection,\ncounting from zero.  Otherwise, use the first occurrence of\nPRESELECT in the collection.  Comparison is first done with\n`equal'.  If that fails, and when applicable, match PRESELECT as\na regular expression.\n\nDEF is for compatibility with `completing-read'.\n\nUPDATE-FN is called each time the candidate list is re-displayed.\n\nWhen SORT is non-nil, `ivy-sort-functions-alist' determines how\nto sort candidates before displaying them.\n\nACTION is a function to call after selecting a candidate.\nIt takes one argument, the selected candidate. If COLLECTION is\nan alist, the argument is a cons cell, otherwise it's a string.\n\nMULTI-ACTION, when non-nil, is called instead of ACTION when\nthere are marked candidates. It takes the list of candidates as\nits only argument. When it's nil, ACTION is called on each marked\ncandidate.\n\nUNWIND is a function of no arguments to call before exiting.\n\nRE-BUILDER is a function transforming input text into a regex\npattern.\n\nMATCHER is a function which can override how candidates are\nfiltered based on user input.  It takes a regex pattern and a\nlist of candidates, and returns the list of matching candidates.\n\nDYNAMIC-COLLECTION is a boolean specifying whether the list of\ncandidates is updated after each input by calling COLLECTION.\n\nEXTRA-PROPS can be used to store collection-specific\nsession-specific data.\n\nCALLER is a symbol to uniquely identify the caller to `ivy-read'.\nIt is used, along with COLLECTION, to determine which\ncustomizations apply to the current completion session." #<bytecode 0x1987a553bed4ebb0>) ("M-x " [eww-buffer-select helm-ff-avfs-directory diary-included-files epg-context-signers haskell-indentation-bird-p dap-ui-breakpoint-log-message nxml-fontify-matcher eww-handle-link gdb-script-syntax-propertize-function reftex-default-bibliography webmail-encode-8bit send-off treemacs-project->path-status gnus-sync haskell-cabal-empty-regexp ns-aliases helm-type-timers-actions treemacs-next-neighbour none-but-delete xsdre-parse-char-class lsp-ui-flycheck chart-axis--eieio-childp lsp-solargraph-references delphi-column-of haskell-mode--inhibit-bracket-inside-comment-or-default semantic--tag-unlink-list-from-buffer lsp-pyls-plugins-flake8-max-line-length ellipses c-standard-font-lock-fontify-region-function bibtex-SuppCollection calc-arccosh tramp-cleanup xsdre-to-symbolic shr-parse-image-data soap-element-namespace-tag--cmacro org-man-fixed-width treemacs-workspace-p--cmacro helm-kill-regexp-as-sexp helm-browse-project-build-buffers-source \' ---5 helm-grep-mode-jump-other-window-forward helm-buffer-run-ediff-merge ---8 treemacs--follow-after-buffer-list-update batch-update-authors treemacs--follow-tag-at-point helm-buffers--get-tab-from-name \0 tramp-gw-open-connection ...] :predicate #f(compiled-function (sym) #<bytecode 0xcf872e21786708c>) :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x))
  ivy-read("M-x " [eww-buffer-select helm-ff-avfs-directory diary-included-files epg-context-signers haskell-indentation-bird-p dap-ui-breakpoint-log-message nxml-fontify-matcher eww-handle-link gdb-script-syntax-propertize-function reftex-default-bibliography webmail-encode-8bit send-off treemacs-project->path-status gnus-sync haskell-cabal-empty-regexp ns-aliases helm-type-timers-actions treemacs-next-neighbour none-but-delete xsdre-parse-char-class lsp-ui-flycheck chart-axis--eieio-childp lsp-solargraph-references delphi-column-of haskell-mode--inhibit-bracket-inside-comment-or-default semantic--tag-unlink-list-from-buffer lsp-pyls-plugins-flake8-max-line-length ellipses c-standard-font-lock-fontify-region-function bibtex-SuppCollection calc-arccosh tramp-cleanup xsdre-to-symbolic shr-parse-image-data soap-element-namespace-tag--cmacro org-man-fixed-width treemacs-workspace-p--cmacro helm-kill-regexp-as-sexp helm-browse-project-build-buffers-source \' ---5 helm-grep-mode-jump-other-window-forward helm-buffer-run-ediff-merge ---8 treemacs--follow-after-buffer-list-update batch-update-authors treemacs--follow-tag-at-point helm-buffers--get-tab-from-name \0 tramp-gw-open-connection ...] :predicate #f(compiled-function (sym) #<bytecode 0xcf872e21786708c>) :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x)
  counsel-M-x()
  funcall-interactively(counsel-M-x)
  call-interactively(counsel-M-x nil nil)
  command-execute(counsel-M-x)
```

**Comment by tesujimath** [comment]
There's not a lot I can do with this. You haven't provided any context at all. Like, what version of Emacs, etc etc etc.

If you want to investigate further and provide some more information, or even a fix, that would be appreciated. I'm interested in keeping this package alive, but do not have time to spend investigating things like this, sorry.

**Comment by stardiviner** [comment]
You're right, I'm using Emacs master branch source code compiled. Version is `Emacs 28.0.50`.
I tried a minimal Emacs config, and try to reproduce this problem, but org-wc works fine. So the problem is on my Org Mode config. I will do `bug-hunter` to figure it out. I will update investiigation result here. Thanks for your package :)

**Comment by stardiviner** [comment]
I did `bug-hunter` on my Emacs inits. I found my configuration is fine on an org-wc.org test file. But command `org-wc-display` report error on my big Org file. I guess some org content caused this issue. Keep update later.

**Comment by stardiviner** [comment]
After testing out many Org elements, I figured Org links caused this problem.
Here is an example:
```org
* This is org-wc

#+begin_src emacs-lisp
(use-package org-wc
  :ensure t)
#+end_src

#+RESULTS[<2019-02-21 16:23:45> 564fb253f1be66f40cad911a7c1bcfce8a017bf9]:
https://www.baidu.com

[[file:org-wc.org]]

[[file:data/code/Microsoft_Recommenders/README.md][README.md]]
```

**Comment by stardiviner** [comment]
I'm using latest version `master` branch source code of Org Mode. version is `9.2.6`.

**Comment by tesujimath** [comment]
Good work isolating this problem! Still unsure when I'll be able to look into this, but appreciate your info, thanks.

**Comment by stardiviner** [comment]
I will look at this code with edebug later when I got time. If I can fix this I will submit PR.

**Comment by andersjohansson** [comment]
I did some updates for org 9.3 in this branch:
 https://github.com/andersjohansson/org-wc/tree/new-org-version
Could be depending on some changes that came already in org 9.2, I don’t quite remember... But maybe it could work to update to org 9.3 and use my branch. Org 9.3 is included in the upcoming Emacs 27. They changed some variable and function names and signatures there.

**Comment by stardiviner** [comment]
@andersjohansson After checking out your branch, and tested. I confirmed this patch works. Thanks for your work. :)

**Comment by stardiviner** [comment]
Gentle ping.... Seems long time, any update on this issue and PR? @tesujimath

**Comment by tesujimath** [comment]
Ah, I'm so sorry for taking a long time over this. I really wanted just to merge all the PRs, but had some issues on the default version of Emacs and org mode from Fedora, and wanted to look into that. Some breakage for me. Is that expected? I'll try to get onto this at the weekend.

**Comment by stardiviner** [comment]
@tesujimath Thanks for taking time. :)

**Comment by andersjohansson** [comment]
Fedora surely packages the currently stable Emacs 26.3 which includes org 9.1 something I think.
I didn't try for more backwards compatibility in PR #25. That commit only works with org 9.3. Stupid perhaps, but it is a bit tricky to support both org 9.3 and <9.3. The biggest problem is that `org-bracket-link-analytic-regexp` is deprecated and replaced by `org-link-bracket-re` which doesn’t match the same groups (it replaces both `org-bracket-link-analytic-regexp` and `org-bracket-link-regexp`). We might be tempted to put a suitable regexp directly in org-wc instead of relying on variables from org, but I believe that the change in the org regexps also reflects the small change in link syntax with org 9.3 (changes in how links are escaped).

So there's no very good solution for compatibility.

**Comment by tesujimath** [comment]
Ok, it sounds like we have to sever links with org < 9.3. I'm ok with that I think.

Again, sorry for the delay in getting to this.

**Comment by tesujimath** [comment]
@stardiviner I have merged the outstanding PRs, so I think this should be fixed if you're running org 9.3

Can you confirm?

Sorry for the loooong delay

**Comment by stardiviner** [comment]
Thanks, confirm fixed.
